File: 과제1/1.cpp
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <fcntl.h>
#include <time.h>

int main(void) {
	int n;
	
	do {
		printf(">> Input the number of numbers to process: ");
		scanf("%d", &n);

		if (n < 2 | n > 30) {
			printf("*Warnning* number of numbers should be between 2 to 30\n\n");
		}
	} while (n < 2 | n > 30);

	int array[n] = {0};

	printf(">> Input the numbers to be processed: \n");

	for (int i = 0; i < n; i++) {
		int k;
		scanf("%d", &k);
		array[i] = k;
	}
	
	time_t start = clock();

// 정렬 
	for (int i = 0; i < n; i++) {
		for (int j = i + 1; j < n; j++) {
			if (array[i] > array[j]) {
				int temp = array[j];
				array[j] = array[i];
				array[i] = temp;
			}
		}
	}

// 소수 개수 반환
	for(int i = 0; i < n-1; i++){
		int count = 0;
		
		while(array[i] == array[i+1]){
			i++;
		}
		
		for(int j = array[i]; j <= array[i+1]; j++){
			int temp = 0;
			for(int k = 2; k < j; k++){
				if((j%k) == 0){
					temp++;
				}
			}
			if(temp == 0){
				count++;
			}
		}
		
		printf("Number of prime numbers between %d, %d: %d\n", array[i], array[i+1], count);
	} 
	
	time_t end = clock();
	printf("\nTotal execution time using C is %lf seconds!", (double)(end - start) / CLOCKS_PER_SEC);
}

File: 과제1/1.py
import time
import math

n = 0

while True:
    try:
        n = int(input(">> Input the number of numbers to process: "))
    except ValueError:
        print("*Warnning* number of numbers should be between 2 to 30")
    if n not in range(2, 30):
        print("*Warnning* number of numbers should be between 2 to 30")
    else:
        break


array = list(map(int, input(">> Input the numbers to be processed : ").split()))

start = time.time()

array.sort()

for i in range(0, n-1):
    count = 0

    if array[i] == array[i+1]:
        continue

    for j in range(array[i] , array[i+1]):
        temp = 0
        for k in range(2, j-1):
            if (j%k) == 0:
                temp += 1
        if temp==0:
            count += 1
    print("Number of prime number between", array[i], ", ", array[i+1], ": ", count)

end = time.time()
print(f"Total execution time using python is {end - start:.5f} seconds")



